app/views/partials/chargeSummary/ChargeSummaryNotCodingOutAndNotPayeSelfAssessment.scala.html

```diff
@@ -43,8 +43,14 @@
             <span class="govuk-tag govuk-tag--red">@messages("chargeSummary.overdue")</span>
         }
         @{
-            if(latePaymentInterestCharge) documentDetail.interestEndDate.getOrElse(throw MissingFieldException("Interest End Date")).toLongDate
-            else documentDetail.documentDueDate.map(_.toLongDate).getOrElse(messages("chargeSummary.na"))
+            if(latePaymentInterestCharge) {
+                documentDetail.interestEndDate.getOrElse(throw MissingFieldException("Interest End Date")).toLongDate
```

Comment:
No exception to be throw in the views pls

Reply:
There are multiple statements like that in that file. TechDebt would be better.